### ISSUE TITLE
Expose `optimizer` and `lr_scheduler` parameters in convenience functions?

### ISSUE THREAD

**Issue by jdeschamps** [CAREamics/careamics#478]
## Problem

Currently, `optimizer` and `lr_scheduler` can only be modified by modifying the configuration directly (or instantiating directly), since they are not exposed in `create_n2v_configuration` convenience function's signature:

https://github.com/CAREamics/careamics/blob/0fa617d7e27c61005274e433a3422be97bcadd7d/src/careamics/config/configuration_factories.py#L748-L767

We could add these parameters, as in some examples we do modify the LR and scheduler settings (see N2V BSD68)

**Comment by Callidior** [comment]
I'm currently already using the following in my scripts:

```python
config = create_n2v_configuration(...)
config.algorithm_config.optimizer.parameters["lr"] = 1e-3
```

So, exposing this directly as a parameter of the factory function would definitely be appreciated.
